### PR TITLE
Provide rowcount for DML queries

### LINKED ISSUE #319: Support row count for DML queries
### Describe the feature

Row count is exposed by the Trino engine when doing DML queries:

```
trino> create table iceberg.default.test AS SELECT 1 a;
CREATE TABLE: 1 row

Query 20230109_192533_00333_sy494, FINISHED, 1 node
Splits: 18 total, 18 done (100,00%)
1,88 [0 rows, 0B] [0 rows/s, 0B/s]

trino> INSERT INTO iceberg.default.test (a) values (2);
INSERT: 1 row
```

This information is not exposed in the Trino Python Client.

It would be nice to expose this information.

DBAPI spec also mentions the following:

> [.rowcount](https://peps.python.org/pep-0249/#rowcount)
> This read-only attribute specifies the number of rows that the last [.execute*()](https://peps.python.org/pep-0249/#id19) produced (for DQL statements like SELECT) or affected (for DML statements like UPDATE or INSERT). [[9]](https://peps.python.org/pep-0249/#id57)
> The attribute is -1 in case no [.execute*()](https://peps.python.org/pep-0249/#id19) has been performed on the cursor or the rowcount of the last operation is cannot be determined by the interface. [[7]](https://peps.python.org/pep-0249/#id55)

### Describe alternatives you've considered

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by mdesmet**
## Description

Resolves #319 

## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
(x) Release notes are required, with the following suggested text:

```markdown
* Provide row count on dbapi's Cursor for DML queries
```

### CODE REVIEW

File: trino/dbapi.py [hunk 1]
```diff
@@ -321,7 +321,8 @@ def rowcount(self):
         operation. For example, the result of a SELECT query is streamed and
         the number of rows is only knowns when all rows have been retrieved.
         """
-
+        if self._query is not None and self._query.update_count is not None:
```
Comment (hovaesco) [inline trino/dbapi.py:324]: The comment above can be updated to reflect the change.

**Review by hovaesco — APPROVED** [review]
LGTM % one comment

File: trino/client.py [hunk 2]
```diff
@@ -290,16 +291,16 @@ def get_roles_values(headers, header):
...
-    def __init__(self, id, stats, warnings, info_uri, next_uri, update_type, rows, columns=None):
-        self.id = id
-        self.stats = stats
-        self.warnings = warnings
-        self.info_uri = info_uri
-        self.next_uri = next_uri
-        self.update_type = update_type
-        self.rows = rows
-        self.columns = columns
+@dataclass
```
Comment (hashhar) [inline trino/client.py:294]: Very nice. ❤️

File: trino/dbapi.py [hunk 3]
```diff
@@ -321,7 +321,8 @@ def rowcount(self):
...
+            return self._query.update_count
         return -1
```
Comment (hashhar) [inline trino/dbapi.py:326]: This is not obvious so maybe a link to https://peps.python.org/pep-0249/#rowcount here would be useful i.e. the -1 is what the DB-API requires.

**Review by hashhar — APPROVED** [review]
LGTM % outdated comment

**Review by damian3031 — APPROVED** [review]

File: tests/integration/test_dbapi_integration.py [hunk 4]
```diff
@@ -1266,10 +1268,56 @@ def test_describe_table_query(run_trino):
...
 def assert_cursor_description(cur, trino_type, size=None, precision=None, scale=None):
     assert cur.description[0][1] == trino_type
     assert cur.description[0][2] is None
     assert cur.description[0][3] is size
     assert cur.description[0][4] is precision
     assert cur.description[0][5] is scale
     assert cur.description[0][6] is None
+
+
+class _TestTable:
```
Comment (hashhar) [inline tests/integration/test_dbapi_integration.py:1306]: Should we have a utils module under test directory? I see some other such classes and methods which we be useful across tests.

**Review by hashhar — APPROVED** [review]
Lgtm. The comment doesn't need to be addressed in this PR, it's just a question.

**Pull request merged**